### ISSUE TITLE
Fix and Create tests for JSON

### ISSUE THREAD

**Issue by ryansurf** [ryansurf/cli-surf#84]
Curretnly, the JSON output fails `curl localhost:8000?json`. We miss this in our tests, so we should write some to catch this!

**Comment by alphonse-xavier-924** [comment]
I am interested in working on this enhancement.

**Comment by ryansurf** [comment]
Hey @alphonse-xavier-924 

I actually assigned this one to my self and am working on it [here](https://github.com/ryansurf/cli-surf/pull/85)

**Comment by alphonse-xavier-924** [comment]
> Hey @alphonse-xavier-924
> 
> I actually assigned this one to myself and am working on it [here](https://github.com/ryansurf/cli-surf/pull/85)
> 
> I'll work on creating some new issues soon. How comfortable are you with python?

@ryansurf  - I'm very comfortable with Python, I'd be happy to take a stab at some of the new issues you create.

FYI - I'm facing issues with docker build on my machine. I think it's related to the bug mentioned here - https://github.com/ryansurf/cli-surf/actions/runs/10233161479/job/28311123053 and the root cause could be an ongoing bug with matplotlib - https://github.com/matplotlib/matplotlib/issues/28551

<img width="707" alt="Error" src="https://github.com/user-attachments/assets/a20edf5c-6ebe-48dc-832b-1d61d0b5a68d">

**Comment by tacaswell** [comment]
The 3.9.1 wheels on windows would, depending on import order, cause segfaults, see https://github.com/matplotlib/matplotlib/issues/28551#issuecomment-2266699619.

We deleted the bad wheels to avoid segfaults (which seemed like a good idea) but failed to anticipate that this would cause downstream projects to start trying to build mpl from source instead of pulling the 3.9.0 wheel.  We have now yanked the whole 3.9.1 release to fix these problems and will get 3.9.2 out as quick as we can.

To prevent this sort of thing in the future (trying to build something from source when you are not prepared to do so) I suggest using `matplotlib --only-binary "matplotlib" ` in your requirements file (which may be a good practice for all wheels that have extensions).

Sorry for the trouble.
